apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.py

```diff
@@ -116,40 +114,45 @@ def render(element_html: str, data: pl.QuestionData) -> str:
                 if (data["panel"] == "submission" or not data["editable"])
                 else "false"
             ),
-            "format": output_format.value,
+            "format": input_format.value,
             "markdown_shortcuts": "true" if markdown_shortcuts else "false",
             "counter": counter.value,
             "counter_enabled": counter != Counter.NONE,
             "clipboard_enabled": clipboard_enabled,
         }
 
-        if source_file_name is not None:
-            if directory == "serverFilesCourse":
-                directory = data["options"]["server_files_course_path"]
-            elif directory == "clientFilesCourse":
-                directory = data["options"]["client_files_course_path"]
-            else:
-                directory = os.path.join(data["options"]["question_path"], directory)
-            file_path = os.path.join(directory, source_file_name)
-            with open(file_path) as f:
-                text_display = f.read()
-        else:
-            text_display = element_text
-
-        html_params["original_file_contents"] = base64.b64encode(
-            text_display.encode("UTF-8").strip()
-        ).decode()
-
         submitted_files = data["submitted_answers"].get("_files", [])
-        submitted_file_contents = [
-            f.get("contents", None)
-            for f in submitted_files
-            if f.get("name", None) == file_name
+        submitted_file = [
```

Comment:
I think you can just call next() on this iterable with a default argument of None to avoid list creation.

Reply:
Done in e2d0c9e3ae4938bad7abde113fa3e885ebeb6cf7.